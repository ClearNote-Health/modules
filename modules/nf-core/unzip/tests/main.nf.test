nextflow_process {

    name "Test Process UNZIP"
    script "../main.nf"
    process "UNZIP"

    tag "modules"
    tag "modules_nfcore"
    tag "unzip"

    test("sarscov2 genome ncbi_taxmap [zip]") {

        when {
            process {
                """
                input[0] = [
                            [ id: 'test' ],
                            file(params.test_data['sarscov2']['genome']['ncbi_taxmap_zip'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot( path(process.out.unzipped_archive[0][1]).list() ).match() } // FIX-ME
            )
        }

    }

    test("sarscov2 genome ncbi_taxmap [zip] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id: 'test' ],
                            file(params.test_data['sarscov2']['genome']['ncbi_taxmap_zip'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.unzipped_archive.collect { file(it[1]).getName() } +
                                process.out.versions ).match() }
            )
        }

    }

}
