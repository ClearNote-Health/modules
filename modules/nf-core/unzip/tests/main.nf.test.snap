{
    "versions": {
        "content": [
            [
                "versions.yml:md5,52c55ce814e8bc9edc5a6c625ed794b8"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T21:16:55.817157215"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "stub"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,52c55ce814e8bc9edc5a6c625ed794b8"
                ],
                "unzipped_archive": [
                    [
                        {
                            "id": "stub"
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,52c55ce814e8bc9edc5a6c625ed794b8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T19:06:19.408027504"
    },
    "unzipped_archive": {
        "content": [
            [
                [
                    {
                        "id": "hello"
                    },
                    [
                        "hello.tar:md5,80c66db79a773bc87b3346035ff9593e"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T21:16:55.802173365"
    }
}